### ISSUE TITLE
Solving `DFTProfile` for binary mixture with `x1 = 1` results in singular matrix

### ISSUE THREAD

**Issue by g-bauer** [feos-org/feos#39]
```python
parameters = PcSaftParameters.from_json(
    ['toluene', 'acetone'], 
    '../parameters/pcsaft/loetgeringlin2018.json'
)
pcsaft = HelmholtzEnergyFunctional.pcsaft(parameters)
vle = PhaseDiagram.binary_vle(pcsaft, 300 * KELVIN, 251)
gamma = [PlanarInterface.from_tanh(pe, 1024, 100 * ANGSTROM, 500 * KELVIN).solve().surface_tension() for pe in vle.states]
```

results in `RuntimeError: The matrix appears to be singular.` for both end-points (pure substances) of the phase diagram.

The above functionality can be done with a `SurfaceTensionDiagram` but would it be helpful to handle `PhaseEquilibrium` inputs for mixtures with `x=1`?

**Comment by prehner** [comment]
The density solver cannot handle components with x=0 because their total chemical potential diverges. Even though a physical solution exists, it cannot be found because the code does not cancel out the diverging ideal gas contributions.

Some thoughts:
- maybe a method to convert `PhaseEquilibrium`s between different component sets could be useful (i.e. add components with n=0 or remove all components where n=0)
- Solving density profiles for one (or more) non-zero components and then adding a zero component can be an interesting case because it gives rise to an inhomogeneous Henry constant and should not suffer from the same problems that the solver has with empty components.

**Comment by prehner** [comment]
This problem is solved in #60. However, any solver with `log=True` will still not converge as it will inevitably call log(0). So a solver has to be passed explicitly. In this case both
```python
DFTSolver().anderson_mixing()
```
and
```python
solver = DFTSolver().picard_iteration(max_iter=3).newton()
```
converge.
